### ISSUE TITLE
feat: Stinger transition on top of DSK

### ISSUE THREAD

**Issue by North-West-Wind** [exeldro/obs-downstream-keyer#43]
I have seen [3](https://obsproject.com/forum/threads/downstream-keyer.142247/page-2#post-532206) [separate](https://obsproject.com/forum/threads/downstream-keyer.142247/page-4#post-572985) [individuals](https://obsproject.com/forum/threads/downstream-keyer.142247/page-5#post-582078) in discussion and [1 review](https://obsproject.com/forum/resources/downstream-keyer.1254/reviews#resource-review-3648) asking for this exact thing, and I too, want this feature.

Currently, DSK overlays on top of any transitions, including Stinger. While there is a "Tie" option, it only ties to the end of a Stinger transition.

**Feature Description**  
Add an option for DSK to be on top or below of transitions.  
Or, when "Tie" is active, tie to the start of the transition instead of the end, allowing synchronized transitions.

**Temporary Solution**  
I stream and run [Firebot](https://firebot.app/) on the side, which is connected to OBS Websocket. I set up 2 event listeners for OBS Scene Transition Started and Ended, and in those 2 events, use [CallVendorRequest](https://github.com/exeldro/obs-downstream-keyer/issues/35) to disable and enable the overlay.

I set DSK to use fade transitions, so it fades out before the transition, and fades in after the transition.

**Comment by dpmesko** [comment]
Yes please! Would love to have topic text banners in a lower third but have them sit underneath global scene transitions (especially when we do stingers). And still use "Exclude Scene" for BRBs, starting/stopping.

@North-West-Wind did you find any further workarounds that let you use "Exclude Scene"?

**Comment by North-West-Wind** [comment]
@dpmesko Yes! By using a combination of [OBS Websocket] and [Advanced Scene Switcher plugin](https://obsproject.com/forum/resources/advanced-scene-switcher.395/) to run macros when Stinger transition starts or ends, together with a fade transition setup for DSK, I can hide the DSK when transition starts and show it again when it ends.

The macros are basically set up like this:  
When Stinger transition starts, send a websocket request to hide the DSK:
```json
{
  "d": {
    "requestId": "dsk-fade-out",
    "requestType": "CallVendorRequest",
    "requestData": {
      "vendorName": "downstream-keyer",
      "requestType": "dsk_select_scene",
      "requestData": {
        "dsk_name": "&{your-dsk-name}",
        "scene": ""
      }
    }
  },
  "op": 6
}
```

When Stinger transition ends, send a websocket request to show the DSK:
```json
{
  "d": {
    "requestId": "dsk-fade-in",
    "requestType": "CallVendorRequest",
    "requestData": {
      "vendorName": "downstream-keyer",
      "requestType": "dsk_select_scene",
      "requestData": {
        "dsk_name": "&{your-dsk-name}",
        "scene": "{your-dsk-scene}"
      }
    }
  },
  "op": 6
}
```
